src/hotspot/share/compiler/compileBroker.cpp

```diff
@@ -515,6 +520,14 @@ CompileQueue* CompileBroker::compile_queue(int comp_level) {
   return nullptr;
 }
 
+CompileQueue* CompileBroker::c1_compile_queue() {
+  return _c1_compile_queue;
+}
+
+CompileQueue* CompileBroker::c2_compile_queue() {
+  return _c2_compile_queue;
+}
+
```

Comment:
Note, `*_compiler_queue` could be `nullptr` if VM is build without C2 or C1 or when run with `-XX:-TieredCompilation` (only C2 is used) or with `-XX:TierdStopAtLevel={1,2,3}` (only C1 is used).

Make sure you check it in JFR event.

Reply:
Thank you!  The JFR event does check for NULL (now nullptr)